libs/hpc-data/src/lib/util.ts

```diff
@@ -359,34 +430,36 @@ export const DATE_FROM_STRING = new t.Type(
   t.identity
 );
 
-const isFormObjectValue = (v: unknown): v is FormObjectValue =>
-  typeof v === 'object' &&
-  !Array.isArray(v) &&
-  v !== null &&
-  Object.keys(v).includes('displayLabel') &&
-  Object.keys(v).includes('value');
-
-/**
- * Accepts a FormObjectValue.
- */
-export const FORM_OBJECT_VALUE = new t.Type<FormObjectValue, FormObjectValue>(
-  'FORM_OBJECT_VALUE',
-  isFormObjectValue,
+export const VALID_DAYJS_DATE = new t.Type<Dayjs, Dayjs>(
+  'VALID_DAYJS_DATE',
+  (u): u is Dayjs => u instanceof Dayjs,
   (v, c) => {
-    if (isFormObjectValue(v)) {
-      return t.success(v);
+    if (isDayjs(v)) {
+      if (v.isValid() && v.year() >= 1950 && v.year() <= 2099) {
+        return t.success(v);
+      }
+      return t.failure(v, c);
     }
     return t.failure(v, c);
   },
   t.identity
 );
 
-export const VALID_DAYJS_DATE = new t.Type<Dayjs, Dayjs>(
-  'VALID_DAYJS_DATE',
-  (u): u is Dayjs => u instanceof Dayjs,
+export const ABORT_SIGNAL = new t.Type<AbortSignal, AbortSignal, unknown>(
+  'AbortSignal',
+  (input: unknown): input is AbortSignal => input instanceof AbortSignal,
+  (input, context) =>
+    input instanceof AbortSignal ? t.success(input) : t.failure(input, context),
+  t.identity
+);
+
+export type AbortSignalType = t.TypeOf<typeof ABORT_SIGNAL>;
+export const YEAR_FROM_STRING = new t.Type<string, string>(
```

Comment:
A bit random place to put this, but I've seen it so many times that I need to comment. Why do you visually group type derived from codec with the next codec? You do this in so many places that it likely is your preferred way of grouping code. Prettier only allows a single space, but I would always group logically similar things (without empty line) and then put empty line to create visual separation from next block. In this case:

```suggestion
export const ABORT_SIGNAL = new t.Type<AbortSignal, AbortSignal, unknown>(
  'AbortSignal',
  (input: unknown): input is AbortSignal => input instanceof AbortSignal,
  (input, context) =>
    input instanceof AbortSignal ? t.success(input) : t.failure(input, context),
  t.identity
);
export type AbortSignalType = t.TypeOf<typeof ABORT_SIGNAL>;

export const YEAR_FROM_STRING = new t.Type<string, string>(
```

Reply:
I would say my preferred way is that they are separated but not close to the next codec. I don't really really like to put type next to const. Anyways I'm not sure I've been very consistent about this statement in this project because I didn't give it to much importance. For the sake of consistency I will modify these spaces in all the hpc-data lib folder like so: 

```ts
export const ABORT_SIGNAL = new t.Type<AbortSignal, AbortSignal, unknown>(
  'AbortSignal',
  (input: unknown): input is AbortSignal => input instanceof AbortSignal,
  (input, context) =>
    input instanceof AbortSignal ? t.success(input) : t.failure(input, context),
  t.identity
);

export type AbortSignalType = t.TypeOf<typeof ABORT_SIGNAL>;

export const YEAR_FROM_STRING = new t.Type<string, string>(
```